src/ui/public/modals/confirm_modal.js

```diff
@@ -23,9 +27,15 @@ module.factory('confirmModal', function ($rootScope, $compile) {
   return function confirmModal(message, customOptions) {
     const defaultOptions = {
       onCancel: noop,
-      cancelButtonText: 'Cancel'
+      cancelButtonText: 'Cancel',
+      onClose: noop,
```

Comment:
I think Stacey's suggestion is the best option for now.

Longer term, I don't think ConfirmModal is the right tool for this use case. I think we should only use ConfirmModal for yes/no questions and create different modals for situations like this one.